polly/lib/Support/GICHelper.cpp

```diff
@@ -143,11 +144,11 @@ static void replace(std::string &str, StringRef find, StringRef replace) {
 }
 
 static void makeIslCompatible(std::string &str) {
-  replace(str, ".", "_");
-  replace(str, "\"", "_");
-  replace(str, " ", "__");
-  replace(str, "=>", "TO");
-  replace(str, "+", "_");
+  replace(str, '.', '_');
```

Comment:
For some reason, the compiler is resolving this replace() to the one I added and not the static one defined above. Hence this change. Also, using std::replace to replace one char with another is likely faster than using string based replacement.

Reply:
The reason is probably that the one in stlextras doesn't require type conversions while the static one above needs to convert to StringRef

Reply:
I see, so if I did an explicit StringRef("") it might resolve correctly. But I feel it might be confusing to a casual reader, so may be the name change is not bad. And also for the single char ones, we get to use the llvm::replace.

Reply:
This works as well:

```
  llvm::replace(str, '.', '_');
  llvm::replace(str, '\"', '_');
  replace(str, StringRef(" "), StringRef("__"));
  replace(str, StringRef("=>"), StringRef("TO"));
  llvm::replace(str, '+', '_');

```

Happy to incorporate it if that is preferable.

Reply:
@arsenm FYI.

Reply:
@kuhar please LMK if we want to do this change.

Reply:
I have any stakes here to be honest, just lurking here :)

Reply:
The implementation of `llvm::replace` looks good to me

Reply:
Updated to use that version. Resolving.